data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/oidc/TokenVerifierImpl.java

```diff
@@ -44,6 +45,13 @@ public Future<JwtClaims> verify(String token, String expectedAudience) {
                 promise -> {
                     // execute blocking, as jose .process() is blocking
 
+                    if (oidcDiscoveryConfig == null) {
+                        promise.fail(
+                                "OIDC discovery config not initialized. This is most likely the case when the pod was started with an invalid OIDC config in place and then later the "
+                                        + FeaturesConfig.KEY_AUTHENTICATION_OIDC
+                                        + " flag was enabled. Restarting the pod should help.");
```

Comment:
Would using the `FileWatcher` to rebuild the config solve the problem of needing to restart the pod?

Reply:
It could. Anyhow changing something fundamental like the OIDC config will probably require some other restarts as well.